### PR TITLE
refactor fallocate03 with new LTP API

### PR DESCRIPTION
**Pull request by ilmanzo**
Tests fallocate() on sparse file for different offsets, with a total of 8 test cases; refactored and simplified code

### CODE REVIEW

File: testcases/kernel/syscalls/fallocate/fallocate03.c [hunk 1]
```diff
@@ -1,277 +1,96 @@
...
+	char fname[NAME_MAX];
 
-	if (fstat(fd, &file_stat) < 0)
-		tst_resm(TFAIL | TERRNO,
-			 "fstat failed while getting block_size");
+	sprintf(fname, "tfile_sparse_%d", getpid());
+	fd = SAFE_OPEN(fname, O_RDWR | O_CREAT, 0700);
+	struct stat file_stat;
 
+	// get fs block size
```
Comment (pevik) [inline testcases/kernel/syscalls/fallocate/fallocate03.c:67]: Please remove this comment, it's obvious.

File: testcases/kernel/syscalls/fallocate/fallocate03.c [hunk 2]
```diff
@@ -1,277 +1,96 @@
...
- ******************************************************************************/
-void file_seek(off_t offset)
-{
-	offset *= block_size;
-	lseek(fd, offset, SEEK_SET);
+	write_data_to_file();
+	// create holes
+	lseek(fd, block_size * (BLOCKS_WRITTEN + HOLE_SIZE_IN_BLOCKS), SEEK_SET);
+	write_data_to_file();
+	lseek(fd, 0, SEEK_SET); // Rewind
```
Comment (pevik) [inline testcases/kernel/syscalls/fallocate/fallocate03.c:76]: Also please remove this comment and the one above (create holes), not much useful.

**Review by pevik — APPROVED** [review]
Generally LGTM.

File: testcases/kernel/syscalls/fallocate/fallocate03.c [hunk 3]
```diff
@@ -1,277 +1,94 @@
...
 
-/*****************************************************************************
- * Create a Hole in the file
- ******************************************************************************/
-void file_seek(off_t offset)
-{
-	offset *= block_size;
-	lseek(fd, offset, SEEK_SET);
+	write_data_to_file();
+	lseek(fd, block_size * (BLOCKS_WRITTEN + HOLE_SIZE_IN_BLOCKS), SEEK_SET);
```
Comment (pevik) [inline testcases/kernel/syscalls/fallocate/fallocate03.c:72]: Please: `SAFE_LSEEK()`.

File: testcases/kernel/syscalls/fallocate/fallocate03.c [hunk 4]
```diff
@@ -1,277 +1,94 @@
...
+	write_data_to_file();
+	lseek(fd, 0, SEEK_SET);
```
Comment (pevik) [inline testcases/kernel/syscalls/fallocate/fallocate03.c:74]: And here: `SAFE_LSEEK()`.

File: testcases/kernel/syscalls/fallocate/fallocate03.c [hunk 5]
```diff
@@ -1,277 +1,94 @@
...
+	{FALLOC_FL_KEEP_SIZE, BLOCKS_WRITTEN},
+	{FALLOC_FL_KEEP_SIZE, BLOCKS_WRITTEN + HOLE_SIZE_IN_BLOCKS / 2 + 1},
+	{FALLOC_FL_KEEP_SIZE, BLOCKS_WRITTEN + HOLE_SIZE_IN_BLOCKS + 2}
 };
 
-int TST_TOTAL = sizeof(test_data) / sizeof(test_data[0]);
-int block_size;
-int buf_size;
+static int block_size;
+static int buf_size;
```
Comment (pevik) [inline testcases/kernel/syscalls/fallocate/fallocate03.c:39]: Am I correct, that there could be just a single variable for these two? Both are just `stat.st_blksize`, right?

Comment (ilmanzo) [reply testcases/kernel/syscalls/fallocate/fallocate03.c:39]: You're correct, they are the same value but I felt it makes the test a bit more readable to keep them logically separated. I'll prepare a better proposal on the next push :)

**Pull request closed without merging**